src/poprox_recommender/components/joiners/rrf.py

```diff
@@ -0,0 +1,41 @@
+from collections import defaultdict
+
+from poprox_concepts import ArticleSet
+from poprox_recommender.lkpipeline import Component
+from poprox_recommender.lkpipeline.types import Lazy
+
+
+class ReciprocalRankFusion(Component):
+    def __init__(self, num_slots: int):
+        self.num_slots = num_slots
+
+    def __call__(self, candidates1: ArticleSet, candidates2: Lazy[ArticleSet]) -> ArticleSet:
+        articles = candidates1.articles
+        article_scores = defaultdict(float)
+        articles_by_id = {}
+
+        i = 1
+        k = 60
+        for article in articles:
```

Comment:
for clarity and error-risk-reduction: this can use `enumerate` rather than manual counters:

```python
for i, article in enumerate(articles, 1):
    ...
```

So can the list below it.

Reply:
I was pondering if vectorizing this computation would be worthwhile; it would be faster, but we would need loops to convert article lists to vectorizable form anyway. It would save dictionary lookups in loops, but might not be worth it. Let's go with it as-is for now and assess performance impact later.